eip7594/src/lib.rs

```diff
@@ -86,29 +95,37 @@ impl From<ThreadCount> for usize {
 /// The context that will be used to create and verify opening proofs.
 #[derive(Debug)]
 pub struct DASContext {
+    #[cfg(feature = "multithreading")]
     thread_pool: Arc<ThreadPool>,
     pub prover_ctx: ProverContext,
     pub verifier_ctx: VerifierContext,
 }
 
+#[cfg(feature = "multithreading")]
 impl Default for DASContext {
     fn default() -> Self {
         let trusted_setup = TrustedSetup::default();
         const DEFAULT_NUM_THREADS: ThreadCount = ThreadCount::Single;
         DASContext::with_threads(&trusted_setup, DEFAULT_NUM_THREADS, UsePrecomp::No)
     }
 }
+#[cfg(not(feature = "multithreading"))]
+impl Default for DASContext {
+    fn default() -> Self {
+        let trusted_setup = TrustedSetup::default();
+
+        DASContext::new(&trusted_setup, UsePrecomp::No)
+    }
+}
 
 impl DASContext {
+    #[cfg(feature = "multithreading")]
     pub fn with_threads(
```

Comment:
rename this to `new` so that when users switch from single to multithread, they just need to modify the number of parameters